### PR TITLE
Add text to full screen spinner when uploading images

### PR DESCRIPTION
**Pull request by boreq**

### CODE REVIEW

**Comment by boreq** [comment]
https://github.com/planetary-social/nos/assets/1935975/fc358266-da0f-41c3-98e7-39e2a87f644d

File: Nos/Views/FullscreenProgressView.swift [hunk 1]
```diff
@@ -6,18 +6,23 @@
...
 
 import SwiftUI
+import SwiftUINavigation
 
 struct FullscreenProgressView: View {
     
     @Binding var isPresented: Bool 
-    
+
+    var text: TextState?
```
Comment (mplorentz) [inline Nos/Views/FullscreenProgressView.swift:15]: `TextState` is a type from `SwiftUINavigation` that adds some features to SwiftUI's default `Text` that we don't need here. In addition the magical extensions we have on `Text` to change the font aren't working for `TextState`. Can you change this to a `String` or `Localized`?
```suggestion
    var text: String?
```

File: Nos/Views/FullscreenProgressView.swift [hunk 2]
```diff
@@ -6,18 +6,23 @@
...
     var hideAfter: DispatchTime?
     
     var body: some View {
         VStack {
             Spacer()
             ProgressView()
                 .foregroundColor(.primaryTxt)
+            if let text {
+                Text(text)
```
Comment (mplorentz) [inline Nos/Views/FullscreenProgressView.swift:24]: The progress wheel and text look a bit too close for my eye. Could you add some padding to the text?
```suggestion
                Text(text).padding()
```

File: Nos/Views/FullscreenProgressView.swift [hunk 3]
```diff
@@ -34,6 +39,6 @@ struct FullscreenProgressView: View {
 
 struct FullscreenProgressView_Previews: PreviewProvider {
     static var previews: some View {
-        FullscreenProgressView(isPresented: .constant(true))
+        FullscreenProgressView(isPresented: .constant(true), text: nil)
```
Comment (mplorentz) [inline Nos/Views/FullscreenProgressView.swift:42]: You can omit `text` here if you want. If you want to add a new preview that includes some text that might be nice also.

**Review by mplorentz — COMMENTED** [review]
I left a few minor comments. Looks pretty good though!

File: Nos/Views/FullscreenProgressView.swift (hunk 2 above)
Comment (boreq) [reply Nos/Views/FullscreenProgressView.swift]: [497e629](https://github.com/planetary-social/nos/pull/492/commits/497e629f9c55f85774cf9b84da2ff069ff80cd3a)

File: Nos/Views/FullscreenProgressView.swift (hunk 1 above)
Comment (boreq) [reply Nos/Views/FullscreenProgressView.swift]: [3eaa652](https://github.com/planetary-social/nos/pull/492/commits/3eaa65270aced8302d8a280b4493971fab99b1da)

File: Nos/Views/FullscreenProgressView.swift (hunk 3 above)
Comment (boreq) [reply Nos/Views/FullscreenProgressView.swift]: [27b18b4](https://github.com/planetary-social/nos/pull/492/commits/27b18b4cf347b473ed12a0687fbb9679a6484e60)

Comment (boreq) [reply Nos/Views/FullscreenProgressView.swift]: https://github.com/planetary-social/nos/assets/1935975/7448b815-9940-4b90-89b7-1474c6f21086

**Review by mplorentz — APPROVED** [review]
Perfection!

**Pull request merged**